### ISSUE TITLE
[Firestore] Cannot set properties of undefined (setting 'fromParams')

### ISSUE THREAD

**Issue by jramosss** [firebase/firebase-admin-node#2498]
### [READ] Step 1: Are you in the right place?

**Cloud Firestore support is provided by the [`@google-cloud/firestore`](https://npmjs.com/package/@google-cloud/firestore) library. Therefore the easiest and most efficient way to get Firestore issues resolved is by directly reporting them at the [nodejs-firestore](https://github.com/googleapis/nodejs-firestore) GitHub repo.**

If you still think the problem is related to the code in this repository, then read on.

  * For issues or feature requests related to __the code in this repository__
    file a Github issue.
  * For general technical questions, post a question on [StackOverflow](http://stackoverflow.com/)
    with the firebase tag.
  * For general Firebase discussion, use the [firebase-talk](https://groups.google.com/forum/#!forum/firebase-talk)
    google group.
  * For help troubleshooting your application that does not fall under one
    of the above categories, reach out to the personalized
    [Firebase support channel](https://firebase.google.com/support/).

### [REQUIRED] Step 2: Describe your environment

  * Operating System version: macOs Sonoma 14.2.1
  * Firebase SDK version: 11.8.0
  * Firebase Product: Firestore
  * Node.js version: 20.0.0
  * NPM version: 9.6.4

#### Steps to reproduce:

What happened? How can we make the problem occur?
This could be a description, log/console output, etc.

You can enable logging for Firestore by including the following line in your code:

```
admin.firestore.setLogFunction(console.log);
```

This will print Firestore logs to the console.

#### Relevant Code:

This is the trace i got by following the past point

Firestore (6.8.0) 2024-03-18T15:41:13.218Z QLnjt [Firestore._retry]: Retrying request that failed with error: TypeError: Cannot set properties of undefined (setting 'fromParams')
    at ClientPool.clientFactory (MYPATH/api/node_modules/firebase-admin/node_modules/@google-cloud/firestore/build/src/index.js:486:42)
    at ClientPool.acquire (MYPATH/api/node_modules/firebase-admin/node_modules/@google-cloud/firestore/build/src/pool.js:103:35)
    at ClientPool.run (MYPATH/api/node_modules/firebase-admin/node_modules/@google-cloud/firestore/build/src/pool.js:217:29)
    at MYPATH/api/node_modules/firebase-admin/node_modules/@google-cloud/firestore/build/src/index.js:1391:30
    at Firestore._retry (MYPATH/api/node_modules/firebase-admin/node_modules/@google-cloud/firestore/build/src/index.js:1251:30)
    at DocumentReader.fetchDocuments (MYPATH/api/node_modules/firebase-admin/node_modules/@google-cloud/firestore/build/src/document-reader.js:89:28)
    at DocumentReader.get (MYPATH/api/node_modules/firebase-admin/node_modules/@google-cloud/firestore/build/src/document-reader.js:52:9)

Basically there's a function that runs before starting my express.js app, that function just calls 
```
myFunction().then(() => {
  app.listen(() => console.log('Server running on port 3000'))
})
```
The function calls 
```
fs.collection('myCollection').get()
```

where fs is exported from other file that exports like this

```
import admin from 'firebase-admin'
export const fs = admin.firestore()
```

**Comment by lahirumaramba** [comment]
```js
import admin from 'firebase-admin'
admin.firestore.setLogFunction(console.log);
```

The above seems to be working when I try to reproduce this issue.

Could you try with the modular imports to see if that helps?
```js
import { setLogFunction } from 'firebase-admin/firestore'
setLogFunction(console.log);
```
